scripts/tools/zap/generate.py

```diff
@@ -268,7 +269,18 @@ def main():
         # Parallel-compatible runs will need separate state
         os.environ["ZAP_TEMPSTATE"] = "1"
 
-    runGeneration(cmdLineArgs.zapFile, cmdLineArgs.zclFile, cmdLineArgs.templateFile, cmdLineArgs.outputDir)
+    # `zap-cli` may extract things into a temporary directory. ensure extraction
```

Comment:
Do we need to do this if `parallel` is false?

And if it's true, should we consider the caller do `zap-cli --version` as @tecimovic suggested before going parallel to avoid this problem?

Reply:
The `zap-cli --version` seems very hackish. This seems to do no significant harm if parallel is not set (I think python still respects tempdir, so we do not lose user configuration if needed). The downside seems to be a 11MB extract on every run instead of potentially re-using the output.

Reply:
Specifically I would like this to be a documented thing that works:

```
time docker run --rm --volume (pwd):/workspace -w /workspace connectedhomeip/chip-build:0.6.17 ./scripts/tools/zap_regen_all.py
```

this does not work because there is no place to add zap-cli --version. You could add commands and bash ... but why.

Reply:
I see.  How long does:

> The downside seems to be a 11MB extract on every run instead of potentially re-using the output.

take in practice?

Reply:
with the change:

```
hyperfine "podman exec -w /workspace bld_vscode ./scripts/tools/zap/generate.py src/controller/data_model/controller-clusters.zap -o zzz_generated/controller-clusters/z
ap-generated"
Benchmark 1: podman exec -w /workspace bld_vscode ./scripts/tools/zap/generate.py src/controller/data_model/controller-clusters.zap -o zzz_generated/controller-clusters/zap-generated
  Time (mean ± σ):      2.813 s ±  0.088 s    [User: 0.037 s, System: 0.025 s]
  Range (min … max):    2.696 s …  2.978 s    10 runs
```

Without the change:

```
hyperfine "podman exec -w /workspace bld_vscode ./scripts/tools/zap/generate.py src/controller/data_model/controller-clusters.zap -o zzz_generated/controller-clusters/zap-generated"
Benchmark 1: podman exec -w /workspace bld_vscode ./scripts/tools/zap/generate.py src/controller/data_model/controller-clusters.zap -o zzz_generated/controller-clusters/zap-generated
  Time (mean ± σ):      2.967 s ±  0.825 s    [User: 0.039 s, System: 0.013 s]
  Range (min … max):    2.648 s …  5.313 s    10 runs

  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.
```

Does not look to make any difference in practice.

Reply:
I can also see it seems that zap varies from time to time for unknown reasons: it should take 2-3 seconds, however the "without the change" I saw it taking 5,9 even 14 seconds.